### PR TITLE
Feat/order book

### PR DESCRIPTION
**Pull request by doylio**
## Motivation
Migrates from Arbitrum Rinkeby contracts to Arbitrum Goerli and enables limit orders

## Changelog

- Update all chain IDs, RPC URLs and Token addresses to use the Arbitrum Goerli deployment
- Remove "Coming Soon" banners and other code blocking limit orders

## Notes
- Subgraphs have not been migrated yet, and some of the rewards contracts haven't been deployed on the testnet yet

### CODE REVIEW

File: src/Addresses.js [hunk 1]
```diff
@@ -15,105 +17,49 @@ const CONTRACTS = {
     USDC: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
   },
 
-  56: {
```
Comment (dospore) [inline src/Addresses.js]: about time we got rid of this stuff

File: src/Api/index.js [hunk 2]
```diff
@@ -1083,6 +1090,7 @@ export async function createIncreaseOrder(
   const shouldWrap = fromETH;
   const triggerAboveThreshold = !isLong;
   const executionFee = getConstant(chainId, "INCREASE_ORDER_EXECUTION_GAS_FEE");
+  console.log("executionFee", executionFee.toString());
```
Comment (dospore) [inline src/Api/index.js]: old log?

File: src/Helpers.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ const SELECTED_NETWORK_LOCAL_STORAGE_KEY = "SELECTED_NETWORK";
 const CHAIN_NAMES_MAP = {
   [MAINNET]: "BSC",
   [TESTNET]: "BSC Testnet",
```
Comment (dospore) [inline src/Helpers.js]:
```suggestion
  [MAINNET]: "Ethereum",
```

File: src/Helpers.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ const SELECTED_NETWORK_LOCAL_STORAGE_KEY = "SELECTED_NETWORK";
...
-  [ARBITRUM_TESTNET]: "Testnet",
+  [ARBITRUM_GOERLI]: "Testnet",
   [ARBITRUM]: "Arbitrum",
   [AVALANCHE]: "Avalanche",
```
Comment (dospore) [inline src/Helpers.js]:
```suggestion
```

File: src/data/Fees.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 export const SECONDS_PER_WEEK = 60 * 60 * 24 * 7;
 
 const FEES = {
-  421611: [],
+  421613: [],
   42161: [
```
Comment (dospore) [inline src/data/Fees.js]: Could update these to network names [ARBITRUM_GOERLI] and [ARBITRUM]

File: src/data/Tokens.js [hunk 6]
```diff
@@ -2,128 +2,12 @@ import { ethers } from "ethers";
...
-    {
-      name: "Binance USD",
-      symbol: "BUSD",
-      decimals: 18,
-      address: "0x3F223C4E5ac67099CB695834b20cCd5E5D5AA9Ef",
-      isStable: true,
-    },
-  ],
-  421611: [
+  421613: [
```
Comment (dospore) [inline src/data/Tokens.js]: same here

File: src/views/Dashboard/AssetDropdown.js [hunk 7]
```diff
@@ -14,7 +14,7 @@ function AssetDropdown({ assetSymbol, assetInfo, trackAction }) {
   let { coingecko, arbitrum, avalanche } = ICONLINKS[chainId][assetSymbol];
   const unavailableTokenSymbols = {
     42161: ["ETH"],
-    421611: ["ETH"],
+    421613: ["ETH"],
```
Comment (dospore) [inline src/views/Dashboard/AssetDropdown.js]: Could name these

**Review by dospore — APPROVED** [review]
Couple of possible additions to the PR but could also be batched into another PR where we remove all reference of AVAX and stuff which I think is probably safer

File: src/Addresses.js [hunk 8]
```diff
@@ -15,105 +17,49 @@ const CONTRACTS = {
...
+    OrderBookReader: "0xf2A0764144bEB244cD04356b5b2078E0ef682FF6",
+    PositionRouter: "0x2d78AE5147D358d803e6306D96db1AB995a4AAF1",
+    PositionManager: "0xb65B42c48DB28e13Cd490f72dDDEEe57eD2072f6",
+
+    ReferralStorage: "0x956BbBe040A2378B63C21011e3c71978d7B75a56",
+    ReferralReader: "0x27ec154C4D6e496027FBED863aC7a0186942Cc09",
+
+    // TODO: Change these once deployed
+    // FeeDistributor: "0x057B99E0112A0D9fDA7503bDc8b6995fA2ff997D",
+    // FeeDistributorReader: "0x55ce0A81b697f1eff846e8a56D3f303A5BE490b1",
```
Comment (dospore) [inline src/Addresses.js]: Commenting these out causes the app to crash when going to referral and rewards page. If you uncomment them it should just error without crashing the app

Comment (dospore) [reply src/Addresses.js]: Resubmitted review

**Review by dospore — CHANGES_REQUESTED** [review]

File: src/Helpers.js (hunk 3 above)
Comment (doylio) [reply src/Helpers.js]: No for some reason `MAINNET` is actually the BSC chain ID

File: src/data/Fees.js (hunk 5 above)
Comment (doylio) [reply src/data/Fees.js]: I tried that, but it throws a weird error when I do.  Something about the order in which the variables are initialized

File: src/components/Exchange/SwapBox.js [hunk 9]
```diff
@@ -1537,6 +1537,11 @@ export default function SwapBox(props) {
       }
     }
 
+    // Limits not enabled for swaps yet
+    if (opt === SWAP && orderOption === LIMIT) {
+      setOrderOption(MARKET);
+    }
+
```
Comment (dospore) [inline src/components/Exchange/SwapBox.js:1544]: should this be in a useEffect?

Comment (dospore) [reply src/components/Exchange/SwapBox.js:1544]: nope, is in a func

**Review by dospore — APPROVED** [review]

**Review by rogue-hacker — APPROVED** [review]

**Pull request merged**